Sources/PackageGraph/PackageGraphLoader.swift

```diff
@@ -150,11 +148,11 @@ private func createResolvedPackages(
         // Make sure these module names are unique in the graph.
         let dependencyModuleNames = dependencies.lazy.flatMap{ $0.modules }.map{ $0.name }
         if let duplicateModules = dependencyModuleNames.duplicates(modules.lazy.map{$0.name}) {
-            errors.append(ModuleError.duplicateModule(duplicateModules.first!))
+            engine.emit(ModuleError.duplicateModule(duplicateModules.first!))
```

Comment:
From an outside perspective - I see `engine.emit` and assume it was the engine emitting something, not a means of ingesting errors to be processed by the engine. I'm not sure I really have a better name, but even something like `engine.addError(...)` would make more semantic sense to me.